packages/block-library/src/navigation/editor.scss

```diff
@@ -77,14 +77,62 @@ $navigation-item-height: 46px;
 	flex-wrap: wrap;
 }
 
-// Vertical aligment.
+// Vertical alignment.
 .wp-block-navigation {
 	min-height: $navigation-height;
 	padding-top: 7px;
 	padding-bottom: 7px;
 	align-items: center;
 }
 
+// Sub menus.
+
+
+// Only display inner blocks when the block is being edited.
+
+.wp-block-navigation {
+	.wp-block-navigation-link {
+		position: relative;
+
+		// Only display inner blocks when the block is being edited.
+		> .block-editor-inner-blocks {
+			display: none;
+		}
+
+		&.is-editing.has-child > .block-editor-inner-blocks {
+			// Box model.
+			display: flex;
+			border-width: 1px;
+			border-style: solid;
+			border-radius: 2px;
+
+			// Position (level 1)
+			position: absolute;
+			z-index: 1;
+
+			// Position (deep levels).
+			.block-editor-inner-blocks {
+				left: calc(100% + 8px); // TODO: use correct variable
+				top: 0;
+			}
+		}
+	}
+
+	// Default / Light styles
+	.wp-block-navigation-link > .block-editor-inner-blocks {
+		color: $light-style-sub-menu-text-color;
+		border-color: $light-style-sub-menu-border-color;
+		background-color: $light-style-sub-menu-background-color;
+	}
+
+	// Dark styles
+	&.is-style-dark .wp-block-navigation-link > .block-editor-inner-blocks {
+		color: $dark-style-sub-menu-text-color;
+		border-color: $dark-style-sub-menu-border-color;
+		background-color: $dark-style-sub-menu-background-color;
+	}
+}
+
```

Comment:
Is these styles going to be portable to the front end and shared?

Reply:
> Are these styles going to be portable to the front end and shared?

Probably(?). I left the code there in case it doesn't set up background color. Guessing that transparent could be a problem. So it will apply the default background-color as long it doesn't already define a custom one.
Something to keep discussing, btw.

Reply:
Something like this:
```css
	// No background color - Default / Light styles
	&:not(.has-background-color) .wp-block-navigation-link > .block-editor-inner-blocks {
		background-color: $light-style-sub-menu-background-color;
	}
```

Reply:
I was talking about all the styles, not just the lines above. I was hoping we'd have a single stylesheet that can apply navigation structure to both the editor and the front end. Colors could be separated so they apply even in situations where you've set a different block style in the future and the structural look is totally different.

Reply:
>  ... was hoping we'd have a single stylesheet that can apply navigation structure to both the editor and the front end.

That makes totally sense to me. Let's find a way to isolate and consume these from both sides.